ourprojects/models.py

```diff
@@ -30,20 +70,114 @@ class Meta(Record.Meta, TracksRun.Meta, TracksUpdates.Meta):
+    uid: str = CharField(
+        unique=True, max_length=12, db_index=True, default=ids.base62_12
+    )
     """Universal id, valid across DB instances."""
     name: str = CharField(db_index=True)
     """Title or name of the Project."""
     abbr: str | None = CharField(max_length=32, db_index=True, unique=True, null=True)
     """A unique abbreviation."""
     url: str | None = URLField(max_length=255, null=True, default=None)
     """A URL to view."""
+    people: Person = models.ManyToManyField(Person, related_name="projects")
+    """People associated with this project."""
+    references: Reference = models.ManyToManyField("Reference", related_name="projects")
+    """References associated with this project."""
     artifacts: Artifact = models.ManyToManyField(
-        Artifact, through="ArtifactProject", related_name="Projects"
+        Artifact, through="ArtifactProject", related_name="projects"
     )
     """Artifacts labeled with this Project."""
 
 
+class Reference(Record, CanCurate, TracksRun, TracksUpdates, ValidateFields):
+    """References such as a publication or document, with unique identifiers and metadata.
+
+    Example:
+        >>> reference = Reference(
+        ...     name="A Paper Title",
+        ...     abbr="APT",
+        ...     url="https://doi.org/10.1000/xyz123",
+        ...     pubmed_id=12345678,
+        ...     doi="10.1000/xyz123",
+        ...     preprint=False,
+        ...     journal="Nature Biotechnology",
+        ...     description="A groundbreaking research paper.",
+        ...     text="A really informative abstract.",
+        ...     published_at=date(2023, 11, 21),
+        ... ).save()
+    """
+
+    class Meta(Record.Meta, TracksRun.Meta, TracksUpdates.Meta):
+        abstract = False
+
+    id: int = models.AutoField(primary_key=True)
+    """Internal id, valid only in one DB instance."""
+    uid: str = CharField(
+        unique=True, max_length=12, db_index=True, default=ids.base62_12
+    )
+    """Universal id, valid across DB instances."""
+    name: str = CharField(db_index=True)
+    """Title or name of the reference document."""
+    abbr: str | None = CharField(
+        max_length=32,
+        db_index=True,
+        unique=True,
+        null=True,
+    )
+    """A unique abbreviation for the reference."""
+    url: str | None = URLField(null=True)
+    """URL linking to the reference."""
+    pubmed_id: int | None = BigIntegerField(null=True, db_index=True)
+    """A PudMmed ID."""
+    doi: str | None = CharField(
+        null=True,
+        db_index=True,
+        validators=[
+            RegexValidator(
+                regex=r"^(?:https?://(?:dx\.)?doi\.org/|doi:|DOI:)?10\.\d+/.*$",
+                message="Must be a DOI (e.g., 10.1000/xyz123 or https://doi.org/10.1000/xyz123)",
+            )
+        ],
+    )
+    """Digital Object Identifier (DOI) for the reference."""
+    preprint: bool = BooleanField(default=False, db_index=True)
```

Comment:
Instead of this, can we add a `Literal` with a bunch of reference types?

Writing this I'm seeing how difficult it is to come up with good vocabulary: `preprint`, `article`,  `internal` all seem wrong. Let's not do it. Nobody will  be happy.

What I think is most important is "external vs. internal" study; most people won't care about it being a preprint or not. But it's also fine to keep that boolean field; booleans are cheap.

Reply:
Should we add a `public` boolean that defaults to "True"? And this can be used to characterize whether it's an internal or external report/paper?

Reply:
Hmm, I think that this is controlled by other software that would actually make the report available. Would I really query for all "non-public" references that I have in my DB? Of course I can craft stories where this matters but I think this would be very rare.